### PR TITLE
Add experimental `build-mode` input

### PR DESCRIPTION
**Pull request by henrymercer**
This PR introduces an experimental `build-mode` input to the `init` Action.  This is currently for internal testing only.

We forward this input to `codeql database init` and add a test to check that the build mode is forwarded to the database as expected, and Java buildless extraction completes successfully.

This PR makes use of the config refactoring implemented in #2107.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: init/action.yml [hunk 1]
```diff
@@ -10,6 +10,20 @@ inputs:
...
+  build-mode:
+    description: >-
+      [Experimental, for internal testing only] The build mode that will be used to analyze the language.
+      This input is only available in single-language analyses.
+
+      Available build modes will differ based on the language being analyzed. One of: 
+
+      - none:      The database will be created without building the source code.
+                   Available for all interpreted languages and some compiled languages.
+      - autobuild: The database will be created by attempting to automatically build the source code.
```
Comment (aeisenberg) [inline init/action.yml:22]: It's unclear here if this still requires an explicit autobuild step.

**Review by aeisenberg — DISMISSED** [review]
I think the action iput description can be clarified, but it's not a blocking change.

File: init/action.yml (hunk 1 above)
Comment (henrymercer) [reply init/action.yml:22]: Good point thanks!  I'll update this.  Right now we do need the explicit autobuild step, but I'm working on a change that will mean no autobuild step is required.

**Review by angelapwen — APPROVED** [review]
🆒 😄

**Pull request merged**